File: BE/src/main/java/khoaluan/BE/service/ReviewService.java
package khoaluan.BE.service;

import khoaluan.BE.request.ReviewRequest;
import khoaluan.BE.response.DataResponse;

public interface ReviewService {
	DataResponse addRating(ReviewRequest reviewRequest);

	DataResponse getRating(Long productId);

	DataResponse isBuy(Long productId, Long userId);
}

File: frontend/src/services/review.service.js
import axios from 'axios';
import authHeader from './auth-header';

const API_URL = 'http://localhost:8080/api/review/';

class ReviewService {
  addRating(reviewRequest) {
    return axios.post(API_URL + 'add-rating', reviewRequest, {});
  }

  getRating(productId) {
    return axios.get(API_URL + 'get-rating?productId=' + productId, {});
  }

  isBuy(productId, useId) {
    return axios.get(
      API_URL + 'is-buy?productId=' + productId + '&userId=' + useId,
      {}
    );
  }
}

export default new ReviewService();
